ui/api/api_test.go

```diff
@@ -0,0 +1,118 @@
+package api
+
+import (
+	"errors"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/dadosjusbr/alba/storage"
+
+	"github.com/labstack/echo/v4"
+	"github.com/steinfletcher/apitest"
+)
+
+var resultDB []storage.Collector = []storage.Collector{
+	{
+		ID:                 "trt13",
+		Entity:             "Tribunal Regional do Trabalho 13ª Região",
+		City:               "João Pessoa",
+		FU:                 "PB",
+		UpdateDate:         time.Date(2009, 10, 17, 20, 34, 58, 651387237, time.UTC),
+		Path:               "github.com/dadosjusbr/coletores/trt13",
+		Frequency:          30,
+		StartDay:           5,
+		LimitMonthBackward: 2,
+		LimitYearBackward:  2018,
+	},
+}
+
+type fakeGetterCollector struct {
+}
+
+func (fake fakeGetterCollector) GetCollectors() ([]storage.Collector, error) {
```

Comment:
Make the fake collector configurable. 

```go
type fakeCollectorsGetter struct{
  resp []storage.Collector
  err error
}

func (fake fakeGetterCollector) GetCollectors() ([]storage.Collector, error) {
   return fake.resp, fake.err
}
```

Dessa forma você pode usar para Nil, NonNil e muitos ouros casos. Não vai precisar criar mais fakes específicos.

Note que, com isso você consegue usar [TableDrivenTests](https://github.com/golang/go/wiki/TableDrivenTests) e tudo vai ficar bonitão. Exemplo [aqui](https://github.com/dadosjusbr/coletores/blob/master/mppe/crawler_test.go#L11)

Reply:
Fiz o fake collector configurável.
Vou adicionar esse recurso de utilização de tabelas de testes numa próxima PR a fim de priorizar o Client para o BD agora.